### PR TITLE
Export characters

### LINKED ISSUE #55593: Expose the characters package to Flutter users for better unicode handling
## Use case

[The characters package](https://pub.dev/packages/characters) was recently introduced to address Dart String's limitations of manipulating extended grapheme clusters (https://github.com/dart-lang/language/issues/34). However, it requires the user to get out of their way to look for it and add it to their Flutter projects. This barrier is likely to cause continued use of the String API in cases where the characters package ought to be used to avoid breaking emojis and non-English text. 

## Proposal

As part of a larger plan to make characters more readily available to writers of text manipulation code, the Flutter framework can export the characters package once it has integrated it into its internal implementation as described in the [TextField Characters Proposal](https://docs.google.com/document/d/1OOFW0PEZf0orBl445YXJ3nfjyKOqZdmeQ6KU3glUh7s/edit). The user-observable effect is that the extension `characters` will become visible when the user does an autocomplete on a String object in their Flutter project.

Alternatively, the Flutter starter app template can always import the characters package. The downside is that the package will only be made available within main.dart unless the user imports it in other files of their project. 

Cc: @justinmc

### PR DESCRIPTION
**Pull request by justinmc**
## Description

We recently landed support for Dart's [characters package](https://pub.dev/packages/characters) in https://github.com/flutter/flutter/pull/59267.  In order to encourage developers to use it in their own projects, what if we export it as a part of Flutter?

Previously, users that wanted to properly handle emojis in Flutter had to install and import the characters package before they could do that. With this PR, they would automatically have access to `string.characters` wherever they're using Flutter.  This makes the barrier much lower for properly handling complex characters.

## Related Issues

Closes https://github.com/flutter/flutter/issues/55593

## Tests

I tested that the characters package is available and works without importing/installing it.

## Breaking Change

None

### CODE REVIEW

File: packages/flutter/lib/widgets.dart [hunk 1]
```diff
@@ -15,6 +15,7 @@
 library widgets;
 
 export 'package:vector_math/vector_math_64.dart' show Matrix4;
+export 'package:characters/characters.dart';
```
Comment (justinmc) [inline packages/flutter/lib/widgets.dart:18]: Is there a better place to export this?

**Comment by mit-mit** [comment]
Exporting this lgtm from a Dart PoV!

File: packages/flutter/lib/widgets.dart (hunk 1 above)
Comment (HansMuller) [reply packages/flutter/lib/widgets.dart:18]: This seems like the best option in lib/*.dart

**Review by HansMuller — APPROVED** [review]
LGTM

**Review by goderbauer — APPROVED** [review]
LGTM

**Comment by justinmc** [comment]
I'm holding off on merging this PR while I fix some logistical problems that caused https://github.com/flutter/flutter/pull/53381 to be reverted.

**Comment by flutter-github-sync** [comment]
Google testing failed...

**Comment by justinmc** [comment]
This or https://github.com/flutter/flutter/pull/59778 appears to have broken google3 despite my CL to add characters to google3.  I must be missing some step.

**Pull request merged**